airbyte-integrations/bases/base-python-test/base_python_test/standard_test.py

```diff
@@ -0,0 +1,111 @@
+"""
+MIT License
+
+Copyright (c) 2020 Airbyte
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import argparse
+import importlib
+import json
+import os.path
+import sys
+from enum import Enum
+
+from .test_iface import StandardSourceTestIface
+
+
+class TestAction(Enum):
+    GET_SPEC = "get_spec"
+    GET_CONFIG = "get_config"
+    GET_CATALOG = "get_catalog"
+    SETUP = "setup"
+    TEARDOWN = "teardown"
+
+
+class StandardSourceTestRunner(StandardSourceTestIface):
+    OUTPUT_FILENAME = "output.json"
+
+    def __init__(self, test):
+        self.test = test
+
+    @staticmethod
+    def write_output(object, path):
```

Comment:
worth noting that not every `object` is going to be JSON serializable. It may be better to take a dict or put a comment saying input must be serializable.

Reply:
thanks. fixed.